improver/utilities/mathematical_operations.py

```diff
@@ -30,9 +30,14 @@
 # POSSIBILITY OF SUCH DAMAGE.
 """Module to contain mathematical operations."""
 
+from typing import Any, List, Optional, Tuple, Union
+
 import iris
 import numpy as np
 import numpy.ma as ma
+from clize.parser import OptionParameter
```

Comment:
Yes, that's a mistake. The automated tool I used to generate annotations required some manual checking. I missed this one (and `typing.Any`).

Reply:
@MoseleyS @benfitzpatrick 

With regards to checking for unused imports we could use [flake8](https://github.com/PyCQA/flake8), for example:

```bash
flake8 improver | grep 'imported but unused'
```

However, given that we're only using pylint to catch errors (with the `-E` flag [here](https://github.com/metoppv/improver/blob/ff658db31364c0bdf3af3940736c260e10544705/.github/workflows/tests.yml#L94)), it might be easier to use flake8 generally as our linter rather than pylint. Running flake8 over `improver` and `improver_tests` takes 5 seconds whereas pylint takes 3 min 26 sec (at least on my workstation).

Reply:
Yep that would be one way to do it! The major attraction of the way we run pylint is the failure on obvious errors - if we could do the same with flake8 that would be good. Not sure how reliable the unused import checking would be, but we could test it. I was actually wondering if our GitHub Hound integration was working correctly - that should be using flake8 and as I understand it should have commented on those PRs...

Reply:
I'm fairly sure that the hound integration is broken - I haven't seen hound comments on PRs for a while. It's unlikely that people have all started writing amazing code and there are no issues being found by hound. The many comments created by hound did used to fill up a lot of vertical space in PR pages.

I'd also be happy for flake8 to get used in the github actions, possibly as a replacement for pylint. I think flake8 could be a better balance of pickiness compared to the current pylint errors -> CI task fails but pylint warnings -> completely ignored approach. The dramatically faster run time of flake8 compared to pylint is also handy.